R/visualSubsetRecode.R

```diff
@@ -0,0 +1,121 @@
+visualSubsetRecode <- function(dat, subsetInfo, ID="ID", toRecodeVal="mci", useGroups=NULL) {
+
+  lapply(list(dat, subsetInfo), checkmate::assert_data_frame, min.rows = 1)
+  checkmate::assert_scalar(ID)
+  stopifnot(ID %in% names(dat))
+  stopifnot(ID %in% names(subsetInfo))
+  checkmate::assert_character(toRecodeVal, len=1)
+  checkmate::assert_character(useGroups, len=1, null.ok=TRUE)
+  if(!is.null(useGroups)) stopifnot(useGroups %in% names(subsetInfo))
+
+  if(any(is.na(subsetInfo))) cli::cli_alert_danger("subsetInfo contains NA values.
+                                              Every subsetInfo row
+                                              containing only a single NA will be
+                                              omitted.", wrap = TRUE)
+  cli::cli_text("")
+  cli::cli_text(paste("--- Begin visual Inspection", Sys.time(), "---"))
+  cli::cli_text("")
+
+  subsetInfo <- unique(na.omit(subsetInfo))
+  datM <- dat
+
+  captureInteraction <- NULL
+
+  i <- 1
+  if(is.null(useGroups)) {
+    ll <- unique(subsetInfo[,ID])[i]
+    nn <- length(unique(subsetInfo[,ID]))
+  } else {
+    pp <- unique(subsetInfo[,useGroups])[i]
+    ll <- unique(subsetInfo[,ID][subsetInfo[,useGroups] %in% pp])
+    nn <- length(unique(subsetInfo[,useGroups]))
+  }
+  res1 <- 0
+
+  while(i <= nn+1) {
+
+    vars <- unique(subsetInfo$datCols[subsetInfo[,ID] %in% ll])
+    sdat <- datM[datM[,ID] %in% ll, vars]
+    cli::cli_inform("Display subset: {ll} (case{?s}) x {vars} (variable{?s}):", wrap = TRUE)
```

Comment:
It might be added how many subsets overall will be inspected.

Reply:
This is a very good idea, I will implement this